manual_correction.py

```diff
@@ -180,112 +308,103 @@ def main():
     args = parser.parse_args()
 
     # Logging level
+    # TODO: how is this actually used?
     if args.verbose:
         coloredlogs.install(fmt='%(message)s', level='DEBUG')
     else:
         coloredlogs.install(fmt='%(message)s', level='INFO')
 
-    # check if input yml file exists
-    if os.path.isfile(args.config):
-        fname_yml = args.config
-    else:
-        sys.exit("ERROR: Input yml file {} does not exist or path is wrong.".format(args.config))
-
-    # fetch input yml file as dict
-    with open(fname_yml, 'r') as stream:
-        try:
-            dict_yml = yaml.safe_load(stream)
-        except yaml.YAMLError as exc:
-            print(exc)
+    # Fetch configuration from YAML file
+    dict_yml = utils.fetch_yaml_config(args.config)
 
     # Curate dict_yml to only have filenames instead of absolute path
     dict_yml = utils.curate_dict_yml(dict_yml)
 
-    # check for missing files before starting the whole process
-    utils.check_files_exist(dict_yml, args.path_in)
+    # Check for missing files before starting the whole process
+    if not args.add_seg_only:
+        utils.check_files_exist(dict_yml, utils.get_full_path(args.path_in))
+
+    suffix_dict = {
+        'FILES_SEG': args.suffix_files_seg,         # e.g., _seg or _label-SC_mask
+        'FILES_GMSEG': args.suffix_files_gmseg,     # e.g., _gmseg or _label-GM_mask
+        'FILES_LABEL': args.suffix_files_label,     # e.g., _labels or _labels-disc
+        'FILES_PMJ': '_pmj'
+    }
 
+    path_out = utils.get_full_path(args.path_out)
     # check that output folder exists and has write permission
-    path_out_deriv = utils.check_output_folder(args.path_out, FOLDER_DERIVATIVES)
+    path_out_deriv = utils.check_output_folder(path_out, args.path_derivatives)
 
     # Get name of expert rater (skip if -qc-only is true)
     if not args.qc_only:
         name_rater = input("Enter your name (Firstname Lastname). It will be used to generate a json sidecar with each "
                            "corrected file: ")
 
     # Build QC report folder name
-    fname_qc = 'qc_corr_' + time.strftime('%Y%m%d%H%M%S')
+    fname_qc = os.path.join(path_out, 'qc_corr_' + time.strftime('%Y%m%d%H%M%S'))
 
     # Get list of segmentations files for all subjects in -path-in (if -add-seg-only)
     if args.add_seg_only:
-        path_list = glob.glob(args.path_in + "/**/*_seg.nii.gz", recursive=True)  # TODO: add other extension
+        path_list = glob.glob(args.path_in + "/**/*" + args.suffix_files_seg + ".nii.gz", recursive=True)
         # Get only filenames without suffix _seg  to match files in -config .yml list
-        file_list = [utils.remove_suffix(os.path.split(path)[-1], '_seg') for path in path_list]
+        file_list = [utils.remove_suffix(os.path.split(path)[-1], args.suffix_files_seg) for path in path_list]
```

Comment:
There are two issues I'm having here for my correction of the manually segmented lesions:
1. If the `suffix_files_seg` is set as `_lesion-manual`, then `path_list` finds all the files correct, but `fname_label` in line 377 below adds another `-manual` suffix. 
2. But, if I give the input for `suffix_files_seg` as `_lesion` (no `-manual` here), then `path_list` doesn't find these files in the first place. 

![Screen Shot 2023-01-24 at 5 08 18 PM](https://user-images.githubusercontent.com/53445351/214432021-69d34ebf-73a5-4edb-b85a-562bb004d071.png)

I think it's because this file assumes that there has an attempt at automatic segmentation already (hence the suffixes _seg and _gmseg, and the manual corrections to those add the suffix -manual to these. But, in my case, I already have the manual lesions which might not be correct so I want to modify them. 

Maybe we should add a different key "FILES_LESION" that does it for lesions separately?

Reply:
Thank you for this feedback! Addressed in the latest commits, especially in https://github.com/spinalcordtoolbox/manual-correction/pull/2/commits/6aae8e96ca47048d2d60f88f65615c80a1d76ccb